controllers/cards.js

```diff
@@ -1,50 +1,45 @@
 const Card = require('../models/card');
+const NotFoundError = require('../errors/not-found-err');
+const ForbiddenError = require('../errors/forbidden-err');
 
-module.exports.createCard = (req, res) => {
+module.exports.createCard = (req, res, next) => {
   const { name, link } = req.body;
   const owner = req.user._id;
   Card.create({ name, link, owner })
     .then((card) => res.status(200).send({ data: card }))
-    .catch((err) => res.status(500).send({ message: err.message }));
+    .catch(next);
 };
 
-module.exports.getCards = (req, res) => {
+module.exports.getCards = (req, res, next) => {
   Card.find({})
     .populate('owner')
     .then((cards) => res.status(200).send({ data: cards }))
-    .catch((err) => res.status(500).send({ message: err.message }));
+    .catch(next);
 };
 
-module.exports.deleteCard = (req, res) => {
-  Card.findById(req.params.cardId)
+module.exports.deleteCard = (req, res, next) => {
+  Card.findById(req.params.cardId).orFail(new NotFoundError('Карточка не найдена'))
     .then((card) => {
-      if (!card) return res.status(404).send({ message: 'Карточка не найдена' });
-      if ((card.owner).toString() !== req.user._id) return res.status(403).send({ message: 'Карта не ваша! Удалить нельзя!' });
-      return Card.remove(card)
-        .then((cardToDelete) => res.send(cardToDelete !== null ? { data: card } : { data: 'Нечего удалять' }))
-        .catch((err) => res.status(500).send({ message: err.message }));
+      if ((card.owner).equals(req.user._id) !== true) throw new ForbiddenError('Карта не ваша! Удалить нельзя!');
+      Card.remove(card)
+        .then((cardToDelete) => res.send(cardToDelete !== null ? { data: card } : { data: 'Нечего удалять' }));
+      return '';
```

Comment:
Можно лучше: использовать единый `.catch()` блок.
В случае с промисами, чтобы встроить новый промис в родительский список нужно внутри `.then()` блока вернуть новый промис:
```suggestion
      return Card.remove(card)
        .then((cardToDelete) => res.send(cardToDelete !== null ? { data: card } : { data: 'Нечего удалять' }));
```